site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/HapiFhirRepository.java

```diff
@@ -32,18 +34,34 @@
         this.fhirContextWrapper = fhirContextWrapper;
     }
 
+    /**
+     * Return the list of all organizations. Note this may include organizations that are
+     * not {uk.ac.ox.ndph.mts.site_service.model.Site}s - caller must filter.
+     * @return all organization instances in the store, might be empty, not null
+     */
+    public Collection<Organization> getOrganizations() {
+        try {
+            // TODO: filter organizations by the extension element that identifies them as sites
+            final Bundle responseBundle = fhirContextWrapper
+                    .search(fhirUri, ORGANIZATION_ENTITY_NAME)
+                    .execute();
```

Comment:
Have you chosen to execute at this class and not of the context on purpose?

Reply:
Yes, since the repository might want to do various searches and already depends on the FHIR API (deals with Bundle, Resources etc.). So this avoids repeating all search methods on the HapiFhirRepository in the FhirContextWrapper.  Given that none of the methods in FhirContextWrapper can be tested effectively, if we add lots of code in the wrapper (that just replicates methods in HapiFhirRepository) we run the risk of getting spuriously bad code coverage. It's just as easy to mock the result of the search() method as it is to mock the whole wrapper, so you can still test HapiFhirRepository.